src/systems/microphone-presence-system.js

```diff
@@ -0,0 +1,58 @@
+const VOLUME_THRESHOLD = 0.00001;
+const UPDATE_FREQUENCY = 1000;
+
+export class MicrophonePresenceSystem {
+  constructor(sceneEl) {
+    this.remountUI = null;
+    this.time = 0;
+    this.sceneEl = sceneEl;
+  }
+
+  tick(dt) {
+    this.time += dt;
+
+    if (this.time > UPDATE_FREQUENCY) {
+      this.time -= UPDATE_FREQUENCY;
+
+      if (
+        !window.APP ||
+        !window.APP.componentRegistry ||
+        !window.APP.hubChannel ||
+        !window.APP.hubChannel.presence ||
+        !this.remountUI
+      ) {
+        return null;
+      }
+
+      const microphonePresences = {};
+
+      const playerInfos = window.APP.componentRegistry["player-info"];
+      if (playerInfos) {
+        for (let i = 0; i < playerInfos.length; i++) {
+          const playerInfo = playerInfos[i];
+          const playerSessionId = playerInfo.playerSessionId;
+          let talking = false;
+          if (playerInfo.isLocalPlayerInfo) {
+            talking = this.sceneEl.systems["local-audio-analyser"].volume > VOLUME_THRESHOLD;
+          }
+          microphonePresences[playerSessionId] = { muted: playerInfo.data.muted, talking };
+        }
+      }
+
+      const networkedAudioAnalysers = window.APP.componentRegistry["networked-audio-analyser"];
+      if (networkedAudioAnalysers) {
+        for (let i = 0; i < networkedAudioAnalysers.length; i++) {
+          const networkedAudioAnalyser = networkedAudioAnalysers[i];
+          if (
+            microphonePresences.hasOwnProperty(networkedAudioAnalyser.playerSessionId) &&
+            networkedAudioAnalyser.volume > VOLUME_THRESHOLD
+          ) {
+            microphonePresences[networkedAudioAnalyser.playerSessionId].talking = true;
+          }
+        }
+      }
+
+      if (this.remountUI) this.remountUI({ microphonePresences });
```

Comment:
I don't think we should use this `remountUI` method. I'd just emit an event when the muted/talking state changes and listen for it in the user list component. Otherwise you're diffing the whole app's UI every frame.

Reply:
this only is updated once per second- is it better to fire an event every second or call remountUI every second? My understanding is the cost of calling remountUI is negligible in the case that the component isn't being rendered (in this case, it's the presence list which is only visible when you have it open) but I'm not familiar enough with react to know for sure. Also in my experience you typically don't want to fire events on repeat in a loop.

Reply:
Or is calling remountUI bad in this case because it's going through ui-root, so it's causing ui-root to update everything? I thought react was smart and knew to only update components that actually had properties that changed.

Reply:
So it's not re-mounting things but it is calling `render()` on the root of the tree. That will require more reconciliation and rerendering than if you listened for the change in the presence list. See this part of the React docs for more details:
https://reactjs.org/docs/reconciliation.html#tradeoffs

I also just don't think that passing this `remountUI` method around is good practice. Here's what I'd do instead:

```js
export default class PresenceList extends Component {
  updateMicrophoneState = () => {
    const microphonePresences = AFRAME.scenes[0].systems["hubs-systems"].microphonePresenceSystem.microphonePresences;
    this.setState({ microphonePresences });
    this.timeout = setTimeout(this.updateMicrophoneState, 1000);
  };

  componentDidMount() {
    this.updateMicrophoneState();
  }

  componentWillUnmount() {
    clearTimeout(this.timeout);
  }
}
```

Alternately:
```js
export default class PresenceList extends Component {
  updateMicrophoneState = () => {
    const microphonePresences = AFRAME.scenes[0].systems["hubs-systems"].microphonePresenceSystem.microphonePresences;
    this.setState({ microphonePresences });
  };

  componentDidMount() {
    document.body.addEventListener("microphone-presence-update", this.updateMicrophoneState);
  }

  componentWillUnmount() {
    document.body.removeEventListener("microphone-presence-update", this.updateMicrophoneState);
  }
}
```

Both of these would only update the part of the UI that needs to be re-rendered.

Reply:
Refactored so that it is no longer in a hubs-system, but rather just a util function (`getMicrophonePresences`). Also updated it so that it will only call this function whenever the presence list is expanded.